sql.go

```diff
@@ -0,0 +1,45 @@
+package ipld_eth_statedb
+
+const (
+	GetContractCodePgStr = `SELECT data FROM public.blocks WHERE key = $1`
+	GetStateAccount      = `SELECT balance, nonce, code_hash, storage_root, removed FROM eth.state_cids
+						INNER JOIN eth.header_cids ON (
+							state_cids.header_id = header_cids.block_hash
+							AND state_cids.block_number = header_cids.block_number
+						)
+						WHERE state_leaf_key = $1
+						AND header_cids.block_number <= (SELECT block_number
```

Comment:
Yeah precisely the ipld-eth-db is populated by the statediffing processes in vdb geth and eth-statediff-service, so it only has records in `state_cids` and `storage_cids` for leaf nodes that were updated at that block height. So if you are trying to find the state for a leaf key at a certain block height, it may not have been updated at that block height so what you want is the latest record below or at that height.